### PR TITLE
feat: Added Node 20 support

### LINKED ISSUE #169: lambda layer: Add support for Node 20
Once AWS makes Node 20 a fully supported Lambda runtime, this repo should publish layers that are compatible.

### PR DESCRIPTION
**Pull request by mrickard**
PR to anticipate Node 20 being introduced as a supported Lambda runtime.

Closes #169 
Closes NR-141663

### CODE REVIEW

**Review by bizob2828 — COMMENTED** [review]
looks like you may have forgot to rebase before adding this. there's a conflict with publish-layers.sh

**Comment by mrickard** [comment]
@bizob2828 I do have to rebase, though I didn't forget: this branch was opened before I removed Node 14 support today.

**Comment by mrickard** [comment]
Noting that I need to add check runs for Node 20.

**Comment by mrickard** [comment]
CI for Node 20 is currently blocked until serverless-offline releases its just-merged support for Node 20: https://github.com/dherault/serverless-offline

**Comment by mrickard** [comment]
@bizob2828 I've rebased this against the wrapper-update PR, and added the new serverless-offline with Node 20 support, so now we're passing.

File: nodejs/index.js
Comment (bizob2828) [inline nodejs/index.js:108]: did you mean to put this stuff in the node 20 PR?

Comment (mrickard) [reply nodejs/index.js:108]: This is rebased off of the performance changes, so I'd imagine that the diff would show that branch's change. I could change the target, or roll back that rebase if we prefer.

Comment (bizob2828) [reply nodejs/index.js:108]: yea. i'd just base this off of the main branch because it's hard to see what's actually changed to support node 20

Comment (mrickard) [reply nodejs/index.js]: Updated to be based off the current main.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**
